### ISSUE TITLE
Enhancement request: override kineticEnergyRequiresForce method to return false in many integrators such as Verlet

### ISSUE THREAD

**Issue by zhi-wang** [openmm/openmm#4751]
I have been performing some computationally expensive calculations and discovered that this behavior is a documented "feature," as noted in Integrator.h

Thanks.

**Comment by peastman** [comment]
Can you give more detail about what you're trying to compute, and what problem you want to solve?  For leapfrog integrators, computing the kinetic energy really does require the forces.  That isn't something you can skip.  It would lead to wrong results.

If you create a CustomIntegrator, you can set whatever expression you want for the kinetic energy, including one that doesn't depend on forces.

**Comment by zhi-wang** [comment]
Oops.

I guess I did not read the documentation carefully. I had the thought that VerletIntegrator was "Velocity-Verlet" but I just found out it was Leap-Frog.

I guess I'll take this chance to ask another question.

I hooked the DFT calculations into MD simulations with OpenMM. What I was trying to do was to validate the conservation of Hamiltonian of my NVE simulations, and to every time-step in the trajectories. I was using "VerletIntegrator" and of course I found out it was twice slower. What integrator would be an ideal substitution to avoid recalculating forces for the kinetic energy in "getState"? Thank you.
